### PR TITLE
feat(operator): add sp1 verifier

### PR DESCRIPTION
**Pull request by entropidelic**

### CODE REVIEW

File: operator/sp1/lib/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[package]
+name = "sp1-verifier-ffi"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+bincode = "1.3.3"
+# sp1-core = { git = "https://github.com/succinctlabs/sp1.git" }
+# sp1-sdk = { git = “https://github.com/succinctlabs/sp1.git”, rev = “v1.0.0-testnet” }
```
Comment (entropidelic) [inline operator/sp1/lib/Cargo.toml]: just a reminder for removing this comments

Comment (taturosati) [reply operator/sp1/lib/Cargo.toml]: Removed on 4c719e0e9772da33371d3692160444d313adb38f

File: .github/workflows/test-sp1.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            sudo rm -rf /opt/ghc
+            sudo rm -rf "/usr/local/share/boost"
+        - uses: actions/checkout@v4
+        - uses: actions/setup-go@v5
+          with:
+            go-version: '1.22'
+            cache: false
+
+        - name: Test sp1 go
+          run: make test-sp1-go-bindings-linux
```
Comment (JuArce) [inline .github/workflows/test-sp1.yml]: missing new line

File: README.md [hunk 3]
```diff
@@ -90,6 +90,7 @@ If `CONFIG_FILE` is not provided, it will use the default config file at `config
 To use the default configuration file, just run:
 
 ```bash
+make build-sp1-macos # or make build-sp1-linux
```
Comment (JuArce) [inline README.md:99]: Missing in normal operator setup flow

File: operator/sp1/sp1.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+/*
+#cgo linux LDFLAGS: ${SRCDIR}/lib/libsp1_verifier.a -ldl -lrt -lm -lssl -lcrypto -Wl,--allow-multiple-definition
+#cgo darwin LDFLAGS: -L./lib -lsp1_verifier
+
+#include "lib/sp1.h"
+*/
+import "C"
+import "unsafe"
+
+const MaxProofSize = 2 * 1024 * 1024
```
Comment (JuArce) [inline operator/sp1/sp1.go:13]: Add a comment with the size in MB

File: operator/sp1/sp1.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+const MaxElfBufferSize = 1024 * 1024
```
Comment (JuArce) [inline operator/sp1/sp1.go:16]: Add a comment with the size in MB

File: task_sender/test_examples/sp1/fibonacci_proof_generator/program/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+[workspace]
+[package]
+version = "0.1.0"
+name = "fibonacci_proof_generator-program"
+edition = "2021"
+
+[dependencies]
+sp1-zkvm = { git = "https://github.com/succinctlabs/sp1.git", rev = "v1.0.0-testnet"  }
```
Comment (JuArce) [inline task_sender/test_examples/sp1/fibonacci_proof_generator/program/Cargo.toml]: missing new line

File: task_sender/test_examples/sp1/fibonacci_proof_generator/script/Cargo.toml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+[package]
+version = "0.1.0"
+name = "fibonacci_proof_generator-script"
+edition = "2021"
+
+[dependencies]
+sp1-sdk = { git = "https://github.com/succinctlabs/sp1.git", rev = "v1.0.0-testnet" }
+
+[build-dependencies]
+sp1-helper = { git = "https://github.com/succinctlabs/sp1.git", rev = "v1.0.0-testnet"  }
```
Comment (JuArce) [inline task_sender/test_examples/sp1/fibonacci_proof_generator/script/Cargo.toml]: missing new line

File: task_sender/test_examples/sp1/fibonacci_proof_generator/script/rust-toolchain [hunk 8]
```diff
@@ -0,0 +1,3 @@
+[toolchain]
+channel = "nightly-2024-04-17"
+components = ["llvm-tools", "rustc-dev"]
```
Comment (JuArce) [inline task_sender/test_examples/sp1/fibonacci_proof_generator/script/rust-toolchain]: missing new line

File: task_sender/test_examples/sp1/fibonacci_proof_generator/script/src/main.rs [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+    // Verify proof and public values
+    client.verify(&proof, &vk).expect("verification failed");
+
+    // Save the proof.
+    let proof_file_path = "../../fibonacci_data/sp1_fibonacci.proof";
+    proof.save(proof_file_path).expect("saving proof failed");
+
+    println!("Successfully generated and verified proof for the program!")
+    // println!("Proof saved to ")
```
Comment (JuArce) [inline task_sender/test_examples/sp1/fibonacci_proof_generator/script/src/main.rs]: remove comment

**Review by JuArce — CHANGES_REQUESTED** [review]

**Review by JuArce — APPROVED** [review]

File: .github/workflows/build.yml [hunk 10]
```diff
@@ -17,9 +17,11 @@ jobs:
         with:
           go-version: '1.22'
           cache: false
-      - name: Build aggregator
-        run: go build aggregator/cmd/main.go
+      - name: Build SP1
```
Comment (entropidelic) [inline .github/workflows/build.yml]:
```suggestion
      - name: Build SP1 bindings
```

File: .github/workflows/test-sp1.yml [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+            sudo rm -rf /usr/share/dotnet
+            sudo rm -rf /opt/ghc
+            sudo rm -rf "/usr/local/share/boost"
+        - uses: actions/checkout@v4
+        - uses: actions/setup-go@v5
+          with:
+            go-version: '1.22'
+            cache: false
+
+        - name: Test sp1 go
```
Comment (entropidelic) [inline .github/workflows/test-sp1.yml]:
```suggestion
        - name: Test SP1 go bindings
```

**Comment by entropidelic** [comment]
lgmt

**Review by uri-99 — APPROVED** [review]
lgtMarian

**Pull request merged**